### DISCUSSION TITLE
Date and/or time object

### DISCUSSION THREAD

**Discussion by blakadder** [HASwitchPlate/openHASP#45] [Ideas]
Object that would display time and date that doesn't rely on sending an mqtt message each second but would use internal clock to display time and date

**Comment by fvanroie** [comment]
The tought has crossed my mind too. I could be implemented as a separate widget, but is actually a label underneath.
Lvgl has timers so it could keep it up-to-date itself without having to chase after it every second.
I'll ponder the idea a bit more and see how it can be accomplished...

One feature it should have is to specify a date/time format string, so users can set it to whatever is appropriate in their region.

**Reply by fvanroie** [reply]
Thinking some more about this, date/time will probably be treated as a sensor or gpio input, so there is a consistent way of handling changing values dynamically.
